ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2

```diff
@@ -1,54 +1,53 @@
    network {{ network_prefix }} area {{ router_ospf.process_ids[process_id].network_prefixes[network_prefix].area }}
-{%             endfor %}
-{%         endif %}
-{%         if router_ospf.process_ids[process_id].bfd_enable is defined and router_ospf.process_ids[process_id].bfd_enable %}
+{%     endfor %}
+{%     if router_ospf.process_ids[process_id].bfd_enable is arista.avd.defined and router_ospf.process_ids[process_id].bfd_enable %}
    bfd default
-{%         endif %}
-{%         if router_ospf.process_ids[process_id].max_lsa is defined %}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].max_lsa is arista.avd.defined %}
    max-lsa {{ router_ospf.process_ids[process_id].max_lsa }}
-{%         endif %}
-{%         if router_ospf.process_ids[process_id].default_information_originate is defined %}
-   default-information originate {% if router_ospf.process_ids[process_id].default_information_originate.always is defined and router_ospf.process_ids[process_id].default_information_originate.always == true %} always{% endif %}
-
-{%         endif %}
-{%         if router_ospf.process_ids[process_id].redistribute.static is defined %}
-{%           if router_ospf.process_ids[process_id].redistribute.static.route_map is defined %}
-   redistribute static route-map {{ router_ospf.process_ids[process_id].redistribute.static.route_map }}
-{%           else %}
-   redistribute static
-{%           endif %}
-{%         endif %}
-{%         if router_ospf.process_ids[process_id].redistribute.connected is defined %}
-{%           if router_ospf.process_ids[process_id].redistribute.connected.route_map is defined %}
-   redistribute connected route-map {{ router_ospf.process_ids[process_id].redistribute.connected.route_map }}
-{%           else %}
-   redistribute connected
-{%           endif %}
-{%         endif %}
-{%     endfor %}
-{% endif %}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].default_information_originate is defined %}
+{%         set default_information_originate_cli = "default-information originate"%}
+{%         if router_ospf.process_ids[process_id].default_information_originate.always is arista.avd.defined(true) %}
+{%             set default_information_originate_cli = default_information_originate_cli ~ " always" %}
+{%         endif %}
+   {{ default_information_originate_cli }}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].redistribute.static is defined %}
+{%         set redistribute_static_cli = "redistribute static" %}
+{%         if router_ospf.process_ids[process_id].redistribute.static.route_map is arista.avd.defined %}
+{%             set redistribute_static_cli = redistribute_static_cli ~ " route-map " ~  router_ospf.process_ids[process_id].redistribute.static.route_map %}
+{%         endif %}
+   {{ redistribute_static_cli}}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].redistribute.connected is defined %}
```

Comment:
Please use `arista.avd.defined` instead

Reply:
We can't at this time as with the current data model we allow for an empty dictionary key "redistribute.connected:" to generate: `redistribute connected`

See test case: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml

Reply:
Got it.